lib/web/mage/requirejs/mixins.js

```diff
@@ -73,51 +91,59 @@ define('mixins', [
     }
 
     /**
-     * Iterativly calls mixins passing to them
+     * Iteratively calls mixins passing to them
      * current value of a 'target' parameter.
      *
      * @param {*} target - Value to be modified.
-     * @param {...Function} mixins
+     * @param {Function[]} mixins - Mixins to be applied.
      * @returns {*} Modified 'target' value.
      */
-    function applyMixins(target) {
-        var mixins = Array.prototype.slice.call(arguments, 1);
-
-        mixins.forEach(function (mixin) {
-            target = mixin(target);
-        });
-
-        return target;
+    function applyMixins(target, mixins) {
+        return mixins.reduce(function (mixedTarget, mixin) {
+            return mixin(mixedTarget);
+        }, target);
     }
 
     rjsMixins = {
 
         /**
          * Loads specified module along with its' mixins.
+         * This method is called for each module defined with "mixins!" prefix
+         * in its name that was added by processNames method.
          *
          * @param {String} name - Module to be loaded.
+         * @param {Function} req - Local "require" function to use to load other modules.
+         * @param {Function} onLoad - A function to call with the value for name.
+         * @param {Object} config - RequireJS configuration object.
          */
         load: function (name, req, onLoad, config) {
-            var path     = getPath(name, config),
-                mixins   = this.getMixins(path),
-                deps     = [name].concat(mixins);
+            var path = getPath(name, config),
+                deps = [name].concat(this.getMixins(path));
 
             req(deps, function () {
-                onLoad(applyMixins.apply(null, arguments));
+                var target = arguments[0],
+                    mixins = [],
+                    index = 1;
+
+                for (; index < arguments.length; index++) {
+                    mixins[index] = arguments[index];
```

Comment:
Array.prototype.slice.call(arguments, 1) could be simpler

Reply:
OP mentions this:

![image](https://user-images.githubusercontent.com/5233399/68916363-3d339400-072b-11ea-8b62-021a6b512c91.png)

For a period of time this was generally a best-practice, as referenced in a [few](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments) [places](https://github.com/davidmarkclements/v8-perf#leaking-and-arrayifying-arguments).

I have no idea if engines have improved optimizations around this in recent years. But I'd bet this isn't something that's going to make or break our performance, so I have no strong preference :)

Reply:
I've just checked it and [it seems](https://jsperf.com/arguments-passing-techniques/1) like passing arguments directly through `apply` is the fastest method of all. I'm going to refactor it.

Reply:
Fixed (reverted).